### PR TITLE
feat(TDOPS-3849): add generic action in HeaderBar component

### PR DESCRIPTION
**Pull request by inna-i**
**What is the problem this PR is trying to solve?**
Button to go "back" to the "Discovery" page for all trial tenants

**What is the chosen solution to this problem?**
- Possibility to add the custom action on the Header bar using **genericAction** props
- HeaderBar.component tests migrated to RTL

**Please check if the PR fulfills these requirements**

- [x] The PR has used `yarn changeset` to request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes/features) And [non-reg](./screenshots.md) done before need review
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Related design/discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by inna-i** [comment]
Generic action props in use:
image

**Review by romainseb — APPROVED** [review]
Thanks for the contribution on this change 🙇🏼

File: packages/components/src/HeaderBar/HeaderBar.test.js [hunk 1]
```diff
@@ -1,58 +1,84 @@
 import React from 'react';
-import { mount, shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
```
Comment (jmfrancois) [inline packages/components/src/HeaderBar/HeaderBar.test.js:2]: :+1:

**Review by jmfrancois — APPROVED** [review]
Thanks for the test in RTL !

File: packages/components/src/HeaderBar/HeaderBar.test.js (hunk 1 above)
Comment (tlnd-mhuchet) [reply packages/components/src/HeaderBar/HeaderBar.test.js:2]: ❤️

**Review by tlnd-mhuchet — APPROVED** [review]

**Pull request merged**